### PR TITLE
Add ActivityInputConverter

### LINKED ISSUE #2674: Quotes added around activity function string parameter in isolated mode
### Description

Using Microsoft.Azure.Functions.Worker.Extensions.DurableTask version 1.1.0, when you pass a string parameter to an activity function (as a parameter with the [ActivityTrigger] attribute), the activity function receives it with quotes added around it.

I have an app that uses Durable Functions isolated mode in .NET 7, and  when I upgraded the Microsoft.Azure.Functions.Worker.Extensions.DurableTask package to 1.1.0 this started happening. Thinking it might be something to do with my code, I used the new project wizard in Visual Studio to create a new Durable Functions isolated project and chose to have it create an orchestration function, so I now had the example "SayHello" app. Running this, it generates the expected `Saying hello to Tokyo`, `Saying hello to Seattle`, `Saying hello to London` messages. But when I updated the NuGet packages so that I'm now using Microsoft.Azure.Functions.Worker.Extensions.DurableTask 1.1.0, it now generates `Saying hello to "Tokyo"`, `Saying hello to "Seattle"`, `Saying hello to "London"`. Also, if you break in the SayHello function, you'll see that the variable "name" doesn't just contain, say, "Tokyo," it contains "\\"Tokyo\\"".

> A clear and concise description of what you expected to happen.

I expected that the activity function would receive the string passed in the CallAcivityAsync call unchanged, as it did in previous versions of the library.

### Actual behavior

The activity function received the string with quotes added around it.

### Relevant source code snippets

You can get this code by just creating the default app with the wizard in Visual Studio, but here are the relevant parts of the code. The orchestration function calls the activity function here:

```
outputs.Add(await context.CallActivityAsync<string>(nameof(SayHello), "Tokyo"));
```

And then the activity function starts with:

```
public static string SayHello([ActivityTrigger] string name, FunctionContext executionContext)
{   
    ...
```

In that function, "name" contains (for the above call) `"Tokyo"`.

### Known workarounds

None that I know of, except for stripping the quotes out of any strings passed to activity functions.

### App Details

Microsoft.Azure.Functions.Worker.Extensions.DurableTask 1.1.0
Azure Functions runtime version 4
Programming language used: C#

### PR DESCRIPTION
**Pull request by jviau**
This PR adds an explicit `ActivityInputConverter` to work around the default Functions converter we were relying on.  Specifically, the default converter special cases a few types (such as string) which we do not want. Adding this converter also has the added benefit of ensuring consistency between orchestrations and activity deserialization.

Another change is to register `DurableTaskClientConverter` directly on `DurableClientAttribute` via `InputConverterAttribute` (this was not available when we first wrote this converter).

### Issue describing the changes in this PR

resolves #2674

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)
* [x] My changes **do not** require any extra work to be leveraged by OutOfProc SDKs
    * [ ] Otherwise: That work is being tracked here: #issue_or_pr_in_each_sdk
* [x] My changes **do not** change the version of the WebJobs.Extensions.DurableTask package
    * [ ] Otherwise: major or minor version updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
* [x] My changes **do not** add EventIds to our EventSource logs
    * [ ] Otherwise: Ensure the EventIds are within the supported range in our existing Windows infrastructure. You may validate this with a deployed app's telemetry. You may also extend the range by completing a PR such as [this one](https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files).
* [x] My changes **should** be added to v3.x branch.
    * [ ] Otherwise: This change only applies to Durable Functions v2.x and **will not** be merged to branch v3.x.

### CODE REVIEW

File: src/Worker.Extensions.DurableTask/HTTP/HttpHeadersConverter.cs
Comment (jviau) [inline src/Worker.Extensions.DurableTask/HTTP/HttpHeadersConverter.cs:1]: This file just has some warning fixes

File: src/Worker.Extensions.DurableTask/DurableTaskExtensionStartup.cs [hunk 1]
```diff
@@ -58,7 +58,6 @@ public override void Configure(IFunctionsWorkerApplicationBuilder applicationBui
 
         applicationBuilder.Services.Configure<WorkerOptions>(o =>
         {
-            o.InputConverters.RegisterAt<DurableTaskClientConverter>(0);
             o.InputConverters.Register<OrchestrationInputConverter>();
```
Comment (davidmrdavid) [inline src/Worker.Extensions.DurableTask/DurableTaskExtensionStartup.cs:61]: why did we remove this line? / why is this safe?

**Review by davidmrdavid — APPROVED** [review]
Looks good! Just leaving a small question

**Comment by davidmrdavid** [comment]
@jviau: can we do a manual test to see if this is addressing [this null-arg issue](https://github.com/Azure/azure-functions-durable-extension/issues/2703)

File: src/Worker.Extensions.DurableTask/DurableTaskExtensionStartup.cs (hunk 1 above)
Comment (jviau) [reply src/Worker.Extensions.DurableTask/DurableTaskExtensionStartup.cs:61]: That line became redundant when this was added:

https://github.com/Azure/azure-functions-durable-extension/pull/2708/files#diff-27ead41f6d3f63e1192a5c044bb8814ae8f417f872c8c1b98b85ae8277a435a2R13-R14

``` CSharp
[InputConverter(typeof(DurableTaskClientConverter))]
[ConverterFallbackBehavior(ConverterFallbackBehavior.Disallow)]
public sealed class DurableClientAttribute : InputBindingAttribute
```

**Comment by jviau** [comment]
> @jviau Jacob Viau FTE: can we do a manual test to see if this is addressing [this null-arg issue](https://github.com/Azure/azure-functions-durable-extension/issues/2703)

Great point - tested it and made adjustments to handle this scenario.

**Comment by prashantagrawalcrowe** [comment]
Currently, I am using **Microsoft.Azure.Functions.Worker.Extensions.DurableTask 1.1.0**, so I wanted to ask in which version this issue would be resolved?

**Pull request merged**